### PR TITLE
[Merged by Bors] - feat: API for multiequalizers in the category of types

### PR DESCRIPTION
**Pull request by joelriou**
In this PR, we give a concrete characterization of limit multiforks in the category of types.

This contribution was created as part of the AIM workshop "Formalizing algebraic geometry" in June 2024.

---

This is used in #14208

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/Shapes/Types.lean [hunk 1]
```diff
@@ -929,4 +931,87 @@ lemma pushoutCocone_inl_eq_inr_iff_of_isColimit {c : PushoutCocone f g} (hc : Is
...
+        | .left i => s.val i
+        | .right j => I.fst j (s.val _)
+      property := by
+        rintro _ _ (_|_|r)
+        · rfl
+        · rfl
+        · exact (s.property r).symm }
+  invFun s :=
+    { val := fun i ↦ s.val (.left i)
+      property := fun r ↦ (s.property (.fst r)).trans (s.property (.snd r)).symm }
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Limits/Shapes/Types.lean:965]:
```suggestion
  toFun s :=
  { val := fun i ↦ match i with
      | .left i => s.val i
      | .right j => I.fst j (s.val _)
    property := by
      rintro _ _ (_|_|r)
      · rfl
      · rfl
      · exact (s.property r).symm }
  invFun s :=
  { val := fun i ↦ s.val (.left i)
    property := fun r ↦ (s.property (.fst r)).trans (s.property (.snd r)).symm }
```

Comment (dagurtomas) [reply Mathlib/CategoryTheory/Limits/Shapes/Types.lean:965]: I don't know if we have a rule for this, feel free to ignore. I just think it looks slightly better this way.

**Review by dagurtomas — APPROVED** [review]
LGTM

File: Mathlib/CategoryTheory/Limits/Shapes/Types.lean (hunk 1 above)
Comment (joelriou) [reply Mathlib/CategoryTheory/Limits/Shapes/Types.lean:965]: Thanks for the reviews! but I do not think that we have any precise rule for `{...}`!

File: Mathlib/CategoryTheory/Limits/Shapes/Types.lean [hunk 2]
```diff
@@ -929,4 +931,87 @@ lemma pushoutCocone_inl_eq_inr_iff_of_isColimit {c : PushoutCocone f g} (hc : Is
...
-end CategoryTheory.Limits.Types
+end Types
+
+section Multiequalizer
+
+variable (I : MulticospanIndex (Type u))
+
+/-- Given a multiequalizer diagram `I : MulticospanIndex (Type u)` in the category
+of types, this is a type which identifies to the sections of the
+functor `I.multicospan.sections`. -/
```
Comment (erdOne) [inline Mathlib/CategoryTheory/Limits/Shapes/Types.lean:942]:
```suggestion
functor `I.multicospan`. -/
```
Also I'd call `I.multicospan` the diagram and `I : MulticospanIndex (Type u)` the index of that diagram.

Comment (joelriou) [reply Mathlib/CategoryTheory/Limits/Shapes/Types.lean]: Thanks! I have changed the phrasing.

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**